components/style/properties/gecko.mako.rs

```diff
@@ -1490,12 +1490,15 @@ fn static_assert() {
     /// This function will also handle scriptminsize and scriptlevel
     /// so should not be called when you just want the font sizes to be copied.
     /// Hence the different name.
+    ///
+    /// Returns true if the inherited size was actually used
     pub fn inherit_font_size_from(&mut self, parent: &Self,
-                                  kw_inherited_size: Option<Au>) {
+                                  kw_inherited_size: Option<Au>) -> bool {
         let (adjusted_size, adjusted_unconstrained_size)
             = self.calculate_script_level_size(parent);
         if adjusted_size.0 != parent.gecko.mSize ||
-           adjusted_unconstrained_size.0 != parent.gecko.mScriptUnconstrainedSize {
+           adjusted_unconstrained_size.0 != parent.gecko.mScriptUnconstrainedSize ||
+           parent.gecko.mSize != parent.gecko.mScriptUnconstrainedSize {
```

Comment:
Can you explain what the addition of this third condition is for?  I can't quite work it out.